app/com/lucidchart/open/cashy/amazons3/S3Client.scala

```diff
@@ -158,4 +160,15 @@ class S3Client {
     }
   }
 
-}
\ No newline at end of file
+}
+
+trait AWSConfig {
+  protected def getAWSCredentials(): AWSCredentials = {
+    try {
+      (new InstanceProfileCredentialsProvider()).getCredentials()
+    } catch {
+      case e: Exception =>
+          (new ProfileCredentialsProvider("/etc/aws/dev-credentials", "default")).getCredentials()
+    }
+  }
```

Comment:
I'd use `new AWSCredentialsProviderChain(new DefaultAWSCredentialsProviderChain, new ProfileCredentialsProvider("/etc/aws/dev-credentials", "default")`.

(Plus, it's better to pass a credentials provider rather than credentials to the client, in the less common case that the credentials are non-static and need to be refreshed.)